contributors/design-proposals/kubelet-authorizer.md

```diff
@@ -0,0 +1,119 @@
+# Scoped Kubelet API Access
+
+Author: Jordan Liggitt (jliggitt@redhat.com)
+
+## Overview
+
+Kubelets are responsible for:
+* creating and updating status of their Node API object
+* updating status of Pod API objects bound to their node
+* creating/deleting "mirror pod" API objects for statically-defined pods running on their node
+* reading Secret and ConfigMap objects referenced by pod specs bound to their node
+
+Currently, kubelets have read/write access to all Node and Pod objects, and read access to all Secret and ConfigMap objects.
+This means that compromising a node gives access to credentials with power to make escalating API calls that affect other nodes.
+
+This proposal limits kubelets' API access using a new node authorizer and admission plugin:
+* Node authorizer
+  * Authorizes requests from nodes using the existing policy rules developed for the `system:node` cluster role
+  * Further restricts secret and configmap requests to only authorize reading objects referenced by pods bound to the node
```

Comment:
I think that would be a breaking change, since static pods can currently use PVCs and configmaps. I suppose we could make it a hard requirement for adopting the node admission plugin, but that might make adoption more difficult.